src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -130,41 +155,43 @@ class InferPythonImportDependencies(InferDependenciesRequest):
     infer_from = PythonSourceField
 
 
-@rule(desc="Inferring Python dependencies by analyzing imports")
-async def infer_python_dependencies_via_imports(
-    request: InferPythonImportDependencies,
-    python_infer_subsystem: PythonInferSubsystem,
-    python_setup: PythonSetup,
-) -> InferredDependencies:
-    if not python_infer_subsystem.imports:
-        return InferredDependencies([])
-
-    _wrapped_tgt = await Get(WrappedTarget, Address, request.sources_field.address)
-    tgt = _wrapped_tgt.target
-    explicitly_provided_deps, parsed_imports = await MultiGet(
-        Get(ExplicitlyProvidedDependencies, DependenciesRequest(tgt[Dependencies])),
-        Get(
-            ParsedPythonImports,
-            ParsePythonImportsRequest(
-                cast(PythonSourceField, request.sources_field),
-                InterpreterConstraints.create_from_targets([tgt], python_setup),
-                string_imports=python_infer_subsystem.string_imports,
-                string_imports_min_dots=python_infer_subsystem.string_imports_min_dots,
-            ),
-        ),
-    )
-
-    resolve = tgt[PythonResolveField].normalized_value(python_setup)
-    owners_per_import = await MultiGet(
-        Get(PythonModuleOwners, PythonModuleOwnersRequest(imported_module, resolve=resolve))
-        for imported_module in parsed_imports
-    )
-
-    merged_result: set[Address] = set()
+def _get_inferred_asset_deps(
+    address: Address,
+    request_file_path: str,
+    assets_by_path: AllAssetTargetsByPath,
+    assets: ParsedPythonAssetPaths,
+    explicitly_provided_deps: ExplicitlyProvidedDependencies,
+) -> Iterator[Address]:
+    for filepath in assets:
+        resource_path = PurePath(request_file_path).parent / filepath
+        inferred_resource_tgts = assets_by_path.resources.get(resource_path, frozenset())
+        inferred_file_tgts = assets_by_path.files.get(PurePath(filepath), frozenset())
```

Comment:
I think it'd be helpful to have a comment explaining why resources only use the `resource_path` and not the full path. But why `file` uses the full path. I originally was envisioning that we look up both mappings with both path types.

I think it makes sense to me why you're doing it this more limited way, but isn't super obvious to me either.

Reply:
Oh hmm that's interesting... Yeah I'll stew on this a bit and make this very clear in a comment. Very good point.

Reply:
Note that dictionary lookups are fast, so it doesn't seem wrong to me to do both lookups?

Reply:
Ok yeah, you made me think about it harder and I'm confident this approach is the right one.

Let me know if the comment needs any tuning.